src/opam_create.ml

```diff
@@ -198,10 +198,9 @@ let add_rules sctx ~dir =
     |> Scope.project
   in
   if Dune_project.generate_opam_files project then begin
-    Local_package.defined_in sctx ~dir
-    |> List.iter ~f:(fun pkg ->
-      let package = Local_package.package pkg in
-      match package.kind with
+    Packages.defined_in sctx ~dir
```

Comment:
Isn't this the same as `Dune_project.packages project`?

Reply:
Not exactly. Suppose we have a structure like this:

```
project/
 dune-project
 package1/
  package1.opam
 package2/
  package2.opam
```

What I have right now is lazier in this case.

Reply:
Yh but that's not supported by Dune. Dune will just see three projects in this case, with both `package1.opam` and `package2.opam` marking the root of two projects

Reply:
Oh, that's right. I switches to use `Dune_project.packages.`.